classes/external/get_data_merge_requests.php

```diff
@@ -0,0 +1,142 @@
+use \tool_mergeusers\merge_request;
+use external_function_parameters;
+use external_multiple_structure;
+use external_single_structure;
+use external_value;
+/**
+ * Version information
+ *
+ * @package     tool
+ * @subpackage  mergeusers
+ * @author      Nicola Vallinoto, Liguria Digitale
+ * @author      Jordi Pujol-Ahulló, SREd, Universitat Rovira i Virgili
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class get_data_merge_requests extends \external_api {
+    /**
+     * Webservice tool_mergeusers_get_data_merge_request parameters
+     *
+     * @return external_function_parameters
+     */
+    public static function execute_parameters(): external_function_parameters {
+        return new external_function_parameters([
+            'mergeusers' => new external_single_structure([
+                'removeuserfield' => new external_value(
+                    PARAM_TEXT, 'Remove user field', VALUE_OPTIONAL),
+                'removeuservalue' => new external_value(
+                    PARAM_RAW, 'Remove user value', VALUE_OPTIONAL),
+                'removeuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the remove user id', VALUE_OPTIONAL),
+                'keepuserfield' => new external_value(
+                    PARAM_TEXT, 'Keep user field', VALUE_OPTIONAL),
+                'keepuservalue' => new external_value(
+                    PARAM_RAW, 'Keep user value', VALUE_OPTIONAL),
+                'keepuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the keep user id', VALUE_OPTIONAL),
+                'id' => new external_value(
+                    PARAM_INT, 'Identifier of the merge request', VALUE_OPTIONAL),
+                'status' => new external_value(
+                    PARAM_INT, 'Status of the merge request', VALUE_OPTIONAL)
+            ])
+        ]);
+    }
+    /**
+     * Return data of the custom queue of the merge request
+     *
+     * @param $mergeusers
+     * @param $removeuserid
+     * @param $removeuservalue
+     * @param $keepuserid
+     * @param $keepuservalue
+     * @param $id
+     * @param $status
+     */
+    public static function execute(array $mergeusers) {
+        global $DB;
+        // Validate all of the parameters.
+        $params = array();
+        /*$params = self::validate_parameters(self::execute_parameters(),
+                                                array('id' => $mergeusers->id));*/
+        $tablemr = merge_request::TABLE_MERGE_REQUEST;
+        $sql = "SELECT
+                    id, removeuserfield, removeuservalue, removeuserid,
+                    keepuserfield, keepuservalue, keepuserid,
+                    timeadded, timemodified, status, retries, log
+                FROM
+                   {".$tablemr."}
+                WHERE
+                     (1=1) ";
+        if ($mergeusers['removeuserfield'] != "" || $mergeusers['removeuserfield'] != null) {
```

Comment:
do you mean `if (isset($mergeusers['removeuserfield']) && !empty($mergeusers['removeuserfield'])) {`

I think that, since this field is optional, it may not be present, and then, generate warnings for accessing an undefined position of an associative array.

Reply:
apply to the other cases below, too, please.